### PR TITLE
EAP - fix variable length field computation

### PR DESCRIPTION
**Pull request by plorinquer**
This PR fixes variable length field computation in EAP_MD5, EAP_TLS and EAP_FAST.

Example:
```
>>> EAP_FAST().show()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "scapy/packet.py", line 923, in show
    return self._show_or_dump(dump, indent, lvl, label_lvl)
  File "scapy/packet.py", line 906, in _show_or_dump
    reprval = f.i2repr(self,fvalue)
  File "scapy/fields.py", line 572, in i2repr
    return x[:self.length_from(pkt)].encode("hex")
  File "scapy/layers/l2.py", line 590, in <lambda>
    XStrLenField('data', '', length_from=lambda pkt: pkt.len - 10 if
pkt.L == 1 else pkt.len - 6)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
```

Additional tests were added in order to prevent computation attempts when field values were `None`.

### CODE REVIEW

File: scapy/layers/l2.py [hunk 1]
```diff
@@ -539,11 +539,12 @@ class EAP_MD5(EAP):
         ByteEnumField("code", 1, eap_codes),
         ByteField("id", 0),
         FieldLenField("len", None, fmt="H", length_of="optional_name",
-                      adjust=lambda p, x: x + p.value_size + 6),
+                adjust=lambda p, x: (x + p.value_size + 6) if p.value_size is not None else 6),
```
Comment (p-l-) [inline scapy/layers/l2.py]: Please keep the correct alignment, it seems broken now (at least in Github).
Can you reverse the test (`if p.value_size is None`)?
Also, do you want `6` when `p.value_size is None` or `x + 6`?
Maybe, in that case, something like `lambda p, x: x + 6 + (0 if p.value_size is None else p.value_size)` would be easier to read. Or even easier: `lambda p, x: x + 6 + (p.value_size or 0)`.

File: scapy/layers/l2.py [hunk 2]
```diff
@@ -539,11 +539,12 @@ class EAP_MD5(EAP):
...
         ByteEnumField("type", 4, eap_types),
-        FieldLenField("value_size", 0, fmt="B", length_of="value"),
+        FieldLenField("value_size", None, fmt="B", length_of="value"),
         XStrLenField("value", '', length_from=lambda p: p.value_size),
-        XStrLenField("optional_name", '', length_from=lambda p: p.len - p.value_size - 6)
+        XStrLenField("optional_name", '',
+                length_from=lambda p: (p.len - p.value_size - 6) if p.len is not None and p.value_size is not None else 0)
```
Comment (p-l-) [inline scapy/layers/l2.py]: Same remarks here.

File: scapy/layers/l2.py [hunk 3]
```diff
@@ -564,7 +565,8 @@ class EAP_TLS(EAP):
         BitField('S', 0, 1),
         BitField('reserved', 0, 5),
         ConditionalField(IntField('tls_message_len', 0), lambda pkt: pkt.L == 1),
-        XStrLenField('tls_data', '', length_from=lambda pkt: pkt.len - 10 if pkt.L == 1 else pkt.len - 6)
+        XStrLenField('tls_data', '',
```
Comment (p-l-) [inline scapy/layers/l2.py]: Please avoid unnecessary changes.

File: scapy/layers/l2.py [hunk 4]
```diff
@@ -587,7 +589,8 @@ class EAP_FAST(EAP):
         BitField('reserved', 0, 2),
         BitField('version', 0, 3),
         ConditionalField(IntField('message_len', 0), lambda pkt: pkt.L == 1),
-        XStrLenField('data', '', length_from=lambda pkt: pkt.len - 10 if pkt.L == 1 else pkt.len - 6)
+        XStrLenField('data', '',
```
Comment (p-l-) [inline scapy/layers/l2.py]: Please avoid unnecessary changes.

File: scapy/layers/l2.py (hunk 1 above)
Comment (plorinquer) [inline scapy/layers/l2.py]: Thanks for the review !

I have fixed the lambda in the `len` field of the `EAP_MD5` class, and rewritten the lambdas in the `EAP_TLS` and `EAP_FAST` as well.

File: scapy/layers/l2.py (hunk 3 above)
Comment (plorinquer) [inline scapy/layers/l2.py]: Maybe I did not understand, but the test on pkt.len is necessary (it must not be `None` in order to perform the computation).

**Comment by plorinquer** [comment]
I noticed I left a change that is not related to this PR (line 1110), but that comes from PR 555.
